### PR TITLE
CVRA Studio - MVP 1: Node discovery

### PR DESCRIPTION
**Pull request by SyrianSpock**
Basic tool performing node discovery over CAN.

Example usage:
```
➤ python node_discovery.py slcan0 --dsdl ../../uavcan_data_types/cvra/
ID    Name                 Status               Health               Uptime    
   45 demo                 OPERATIONAL          OK                   0:00:02   
  120 test                 INITIALIZATION       OK                   0:00:03   
```

The features are very basic, but it shows the idea of what we want to build and how: Model View Controller architecture and common UAVCAN interface handling spin/lock.

### CODE REVIEW

**Comment by SyrianSpock** [comment]
Not sure if I should also add this script that spins a dummy node on UAVCAN, useful to debugging

```python
import argparse
import uavcan

def parse_args():
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("interface", help="Serial port or SocketCAN interface")
    parser.add_argument("id", help="UAVCAN node ID", type=int)
    parser.add_argument("name", help="UAVCAN node name", type=str)

    return parser.parse_args()

def main():
    args = parse_args()

    node_info = uavcan.protocol.GetNodeInfo.Response(name=args.name)
    node = uavcan.make_node(args.interface, node_id=args.id, node_info=node_info)

    while True:
        try:
            node.spin(1)
        except uavcan.UAVCANException as ex:
            print('Node error:', ex)

if __name__ == '__main__':
    main()
```

Usage: `python node.py slcan0 120 test`

File: tools/studio/node_discovery.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+class UavcanNode:
+    def __init__(self, interface, node_id):
+        self.handlers = []
+        self.node_lock = threading.RLock()
+        self.node = uavcan.make_node(interface, node_id=node_id)
+
+    def subscribe(self, topic, callback):
+        self.handlers.append(self.node.add_handler(topic, callback))
+
+    def service_call(self, request, node_id, callback):
```
Comment (antoinealb) [inline tools/studio/node_discovery.py]: Maybe name it `request`, since it wraps UAVCAN's request method ?

File: tools/studio/node_discovery.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import time
+import uavcan
+
+class UavcanNode:
+    def __init__(self, interface, node_id):
+        self.handlers = []
+        self.node_lock = threading.RLock()
+        self.node = uavcan.make_node(interface, node_id=node_id)
+
+    def subscribe(self, topic, callback):
```
Comment (antoinealb) [inline tools/studio/node_discovery.py]: Same as below, we could name it add_handler.

File: tools/studio/node_discovery.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        print(formatted_line.format("ID", "Name", "Status", "Health", "Uptime"))
+        for node in nodes:
+            name = nodes[node]['name']
+            status = nodes[node]['status']
+            print(formatted_line.format(node, name, self._display_status(status.mode),
+                                        self._display_health(status.health),
+                                        self._display_uptime(status.uptime_sec)))
+        self._previous_print_len = 1 + len(nodes)
+
+    def _delete_previous_print(self):
```
Comment (antoinealb) [inline tools/studio/node_discovery.py]: How does it delete code ? Maybe drop a comment.

**Pull request merged**